File: leetcode_java/161. One Edit Distance.java
class Solution {
    public boolean isOneEditDistance(String s, String t) {
        int dif = t.length() - s.length(), i = 0;
        if (dif < 0) return isOneEditDistance(t, s);
        if (dif > 1) return false;
        while (i < s.length() && s.charAt(i) == t.charAt(i)) i++;
        if (i == s.length()) return dif > 0;
        if (dif == 0) i++;
        while (i < s.length() && s.charAt(i) == t.charAt(i + dif)) i++;
        return i == s.length();
    }
}

File: leetcode_python/161. One Edit Distance.py
class Solution:
    def isOneEditDistance(self, s: str, t: str) -> bool:
        i, dif = 0, len(t) - len(s)
        if dif < 0: s, t, dif = t, s, -dif
        if dif > 1: return False
        while i < len(s) and s[i] == t[i]: i += 1
        if i == len(s): return bool(dif)
        if not dif: i += 1
        while i < len(s) and s[i] == t[i + dif]: i += 1
        return i == len(s)

s = Solution()
print(s.isOneEditDistance("acb","abc"))
